_posts/2020-08-07-g2p.md

```diff
@@ -392,40 +406,215 @@ You can also use abbreviations like this in the G2P studio by writing them in th
+
+Below is an example of the embeddable web component in Danish.
+
+{% include readalong.html text="/assets/ras/output.xml" alignment="/assets/ras/output.smil" audio="/assets/ras/output.wav" %}
 
 ### Convertextract
-[Convertextract]({{ "convertextract-app" | absolute_url }})
+
+Convertextract is a tool that lets you use `g2p` mappings to convert text inside a variety of different documents. Have a look at the [convertextract repo readme](https://github.com/roedoejet/convertextract) or at Fineen Davis' [blog post]({{ "convertextract-app" | absolute_url }}) about the Convertextract GUI.
 
 ### Advanced: use in Python
 
-...more to come...
+#### Installation
+
+We recommend installing `g2p` using [pip](https://en.wikipedia.org/wiki/Pip_(package_manager)):
+
+```python
+pip install g2p
+```
+
+If you are going to be creating your own mapping or editing `g2p` in any way, you must first fork and clone `g2p`:
+
+```bash
+git clone https://github.com/YourGitHubUsername/g2p.git
+```
+
+and then do an *editable* pip installation:
+
+```bash
+cd g2p && pip install -e .
+```
+
+#### Usage
+
+Using `g2p` within Python can be done, either programatically using the `make_g2p` function:
+
+```python
+>>> from g2p import make_g2p
+>>> transducer = make_g2p('dan', 'eng-arpabet')
+>>> transducer('hej').output_string
+'HH EH Y'
+```
+
+#### Command Line Interface
+
+You can also use `g2p` from the command line. The basic command for conversions is:
+
+```bash
+g2p convert <input_text> <in_lang> <out_lang>
+```
+
+So in practice:
+
+```bash
+$ g2p convert hej dan eng-arpabet
+HH EH YY
+```
 
 ## Advanced: contributing your rules to the main `g2p` library
 
-You've written some cool rules and you want to contribute, that's awesome! 
+You've written some cool rules and you want to contribute, that's awesome! There are lots of benefits to contributing your mapping to `g2p`. First of all, once your mapping is accepted, you'll have it available and live on [G2P Studio](https://g2p-studio.herokuapp.com). Second, once the next version of `g2p` is release with your mapping, it will be automatically built in to the [Convertextract](#convertextract) library. Third, if your mapping is between a language's writing system and the IPA, you can also get ReadAlongs support for your language.
+
+So, you write your mapping once, and you get three things for free (G2P studio, convertextract and readalongs). Here's how:
+
+1. Fork g2p
+2. Add a folder for your language using the appropriate [ISO 639.3 code](https://en.wikipedia.org/wiki/List_of_ISO_639-3_codes) to `g2p/mappings/langs`
+3. Add a config.yaml file as described [here](#mapping-configuration)
+4. Add your mapping in that same folder (`g2p/mappings/langs/<yourlang>`)
+5. If your mapping is for an IPA mapping, you can optionally [generate the mapping](#generate-your-mapping-between-your-languages-ipa-and-english-ipa) between your language and English IPA.
```

Comment:
you have to run `g2p update` before and after this step, not just after.

Reply:
good point, thanks @joanise !